src/routes/users/updateUser.test.ts

```diff
@@ -0,0 +1,64 @@
+import { FastifyInstance, RouteOptions } from "fastify";
+import createServer from "../../server";
+import updateUser from "./updateUser";
+import UserModel from "../../models/user";
+
+describe("PUT /user", () => {
+  let server: FastifyInstance;
+  beforeAll(() => {
+    server = createServer({ authRoute: [updateUser] as RouteOptions[] });
+  });
+
+  it("should update a user correctly", async () => {
+    const newUser = {
+      firstName: "Joe",
+      lastName: "Bloggs",
+      CreatedBy: "ADMIN_TEST",
+      ModifiedBy: "ADMIN_TEST",
+      CreatedDate: new Date(),
+      ModifiedDate: new Date(),
+    };
+    await UserModel.query().insert(newUser);
+    const userId = 1;
+    const newUserFirstName = "John";
+    const newUserLastName = "Doe";
+    const newUserAreaId = 1;
+
+    await UserModel.query().updateAndFetchById(userId, {
+      firstName: newUserFirstName,
+      lastName: newUserLastName,
+      areaId: newUserAreaId,
+    });
+
+    const response = await server.inject({
+      method: "PUT",
+      url: `/user/${userId}`,
+      payload: {
+        firstName: "John",
+        lastName: "Doe",
+        areaId: 1,
+      },
+    });
+
+    expect(response.statusCode).toEqual(200);
+    const updatedUser = JSON.parse(response.payload);
+    expect(updatedUser.firstName).toEqual(newUserFirstName);
```

Comment:
Here, we are expecting an object. So instead of testing each value expected, we can use [objectContaining](https://jestjs.io/docs/expect#expectobjectcontainingobject)